src/protocol/ascii.rs

```diff
@@ -148,58 +162,95 @@ impl AsciiProtocol<Stream> {
         value: V,
         options: &Options,
     ) -> Result<bool, MemcacheError> {
-        check_key_len(key)?;
+        Ok(self.stores(command, Some((key, value)), options)?)
+    }
+
+    fn stores<V: ToMemcacheValue<Stream>, K: AsRef<str>, I: IntoIterator<Item = (K, V)>>(
+        &mut self,
+        command: StoreCommand,
+        entries: I,
+        options: &Options,
+    ) -> Result<bool, MemcacheError> {
         if command == StoreCommand::Cas {
             if options.cas.is_none() {
                 Err(ClientError::Error(Cow::Borrowed(
                     "cas_id should be present when using cas command",
                 )))?;
             }
         }
+
         let noreply = if options.noreply { " noreply" } else { "" };
-        if options.cas.is_some() {
-            write!(
-                self.reader.get_mut(),
-                "{command} {key} {flags} {exptime} {vlen} {cas}{noreply}\r\n",
-                command = command,
-                key = key,
-                flags = value.get_flags(),
-                exptime = options.exptime,
-                vlen = value.get_length(),
-                cas = options.cas.unwrap(),
-                noreply = noreply
-            )?;
-        } else {
-            write!(
-                self.reader.get_mut(),
-                "{command} {key} {flags} {exptime} {vlen}{noreply}\r\n",
-                command = command,
-                key = key,
-                flags = value.get_flags(),
-                exptime = options.exptime,
-                vlen = value.get_length(),
-                noreply = noreply
-            )?;
-        }
+        let mut sent_count = 0;
+
+        {
```

Comment:
Why this block?

Reply:
I added that block to drop the ``reader`` var using the scope, but on further reflection, the extra indentation is obviously not good. I'm fixing it.